tests/api/shelves/by_owners.test.js

```diff
@@ -22,62 +23,85 @@ describe('shelves:by-owners', () => {
 
     it('should get a public shelf', async () => {
       const { shelf } = await createShelf()
-      shelf.listing.should.equal('public')
+      shelf.visibility.should.deepEqual([ 'public' ])
       const res = await publicReq('get', `${endpoint}&owners=${shelf.owner}`)
-      res.shelves.should.be.ok()
+      res.shelves[shelf._id].should.be.ok()
     })
   })
 
-  describe('listing:private', () => {
+  describe('visibility:private', () => {
     it('should return user shelf', async () => {
-      const { shelf } = await createShelf(null, { listing: 'private' })
+      const { shelf } = await createShelf(null, { visibility: [] })
       const user = await getUser()
       const res = await authReq('get', `${endpoint}&owners=${user._id}`)
-      _.map(res.shelves, _.property('_id')).should.containEql(shelf._id)
+      res.shelves[shelf._id].should.be.ok()
     })
 
     it('should not return private shelves', async () => {
-      const { shelf } = await createShelf(getUserB(), { listing: 'private' })
+      const { shelf } = await createShelf(getUserB(), { visibility: [] })
       const user = await getUserB()
       const res = await authReq('get', `${endpoint}&owners=${user._id}`)
-      const resIds = Object.keys(res.shelves)
-      resIds.should.not.containEql(shelf._id)
+      should(res.shelves[shelf._id]).not.be.ok()
     })
 
     it('should not return friends private shelves', async () => {
       const friendA = await createUser()
       const friendB = await createUser()
       await makeFriends(friendA, friendB)
 
-      const { shelf } = await createShelf(friendB, { listing: 'private' })
+      const { shelf } = await createShelf(friendB, { visibility: [] })
       const { _id: friendBId } = await friendB
       const res = await authReq('get', `${endpoint}&owners=${friendBId}`)
```

Comment:
looks wrong since the beginning, luckily its still green after adding suggestion below
```suggestion
      const [ userA, userB ] = await getTwoFriends()
      const { shelf } = await createShelf(userA, { visibility: [] })
      const res = await customAuthReq(userB, 'get', `${endpoint}&owners=${userA._id}`)
      should(res.shelves[shelf._id]).not.be.ok()
```

Reply:
it was green on my side, but the `getTwoFriends` implementation should be more efficient anyway, so done in e4edb029